x-pack/plugins/observability_solution/logs_explorer/public/state_machines/logs_explorer_controller/src/services/data_view_service.ts

```diff
@@ -43,7 +43,7 @@ export const changeDataView =
     // when updating the control panels.
     const nextDataView = await dataViews.get(context.datasetSelection.toDataviewSpec().id, false);
     if (nextDataView.id) {
-      await discoverStateContainer.actions.onChangeDataView(nextDataView.id);
+      await discoverStateContainer.actions.onChangeDataView(nextDataView);
```

Comment:
**note**: passing directly the retrieved data view instance avoids re-fetching the same data view as would internally do this method if it receives a data view id.